src/spikeinterface/core/base.py

```diff
@@ -366,27 +366,24 @@ def to_dict(
                     new_kwargs[name] = transform_extractors_to_dict(value)
 
             kwargs = new_kwargs
-        class_name = str(type(self)).replace("<class '", "").replace("'>", "")
+
+        module_import_path = self.__class__.__module__
+        class_name_no_path = self.__class__.__name__
+        class_name = f"{module_import_path}.{class_name_no_path}"  # e.g. 'spikeinterface.core.generate.AClass'
         module = class_name.split(".")[0]
-        imported_module = importlib.import_module(module)
 
-        try:
-            version = imported_module.__version__
-        except AttributeError:
-            version = "unknown"
+        imported_module = importlib.import_module(module)
+        spike_interface_version = getattr(imported_module, "__version__", "unknown")
```

Comment:
```suggestion
        module_version = getattr(imported_module, "__version__", "unknown")
```
@h-mayorquin this could also be something external/on top of to SpikeInterface, like SpikeForest

Reply:
Ah! good to know.

Reply:
OK, modified this and improved the docstring a wee more.